cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -452,7 +452,7 @@ def add_button(self, **kwargs):
         view.add.click()
         view = self.create_view(DetailsCatalogItemView, wait=5)
         view.flash.assert_no_error()
-        return text
+        return hover
```

Comment:
Thanks so much for tackling a test fix! Things tend to get complex when the parameters and return values for framework code changes.

The test that has been modified along with this change is not the only use of this method within the framework, and the other uses are using this return value to then make other calls - like to `button_exists` below, which takes the text, not the hover.

Check out `cfme/tests/automate/custom_button/test_catalog_custom_button.py::test_catalog_item_copy_with_custom_buttons`, which is using this return, and kind of highlights how there is duplicated coverage for the fixed test.

A small scope fix would be to actually pass text/hover attributes to the add_button() call in the failing test.  That way you can use the hover attribute that you passed to validate the flash message.  You can also consider just removing the test completely, as the behavior is covered by the above test function.

I think the correct fix, from a framework perspective, is that this method should be returning an instance of `cfme.automate.button.DefaultButton`, and that all callers should be updated to handle that return.  That way all of the attributes are available for further use and validation.